### DISCUSSION TITLE
Adaptive ANM should handle reduce and slice

### DISCUSSION THREAD

**Discussion by jamesmkrieger** [prody/ProDy#2058] [Archived Issues]
We won't have this for the current release, but it would be a useful feature to have to account for environment. For example, the effect of membrane, which differs between models, or an auxiliary subunit that is in one of the two structures.

**Comment by SHZ66** [comment]
This feature needs to be implemented carefully as it cannot break the current implementation where dummy atoms are removed from the ANM calculation. These dummies cannot be reduced, as they have the same coordinates, and reducing their Hessian terms will likely cause an error. A correct implementation should remove (trim) the dummies first, and then reduce. In the meantime, it should retain the flexibility that users can choose which way they want to treat the system (trim, slice, or reduce).

**Comment by jamesmkrieger** [comment]
I think this is actually handled through calcENM but we need to check
